### ISSUE TITLE
Custom event handler for the Close On Escape 'keydown' event 

### ISSUE THREAD

**Issue by Oliviero50** [Flyer53/jsPanel4#162]
**Is your feature request related to a problem? Please describe.**
It is currently not possible to override the event handler that handles the closeOnEscape 'keydown' event. (the event handler is an anonymous function)
Current implementation:
``` javascript
var jsPanel = {
    ...
    closeOnEscape: function () {
        document.addEventListener('keydown', function (e) {
            if (e.key === 'Escape' || e.code === 'Escape' || e.key === 'Esc') {
                jsPanel.getPanels(function () {
                    return this.classList.contains('jsPanel');
                }).some(function (item) {
                    if (item.options.closeOnEscape) {
                        item.close(null, true);
                        return true;
                    }
                    return false;
                });
            }
        }, false);
    }(),
    ...
};
```

**Describe the solution you'd like**
The simplest solution would be to have a property jsPanel.customCloseOnEscHandler that is called instead of the default implementation, if it is set.
Possible solution:
``` javascript
var jsPanel = {
    version: '4.11.4',
    ...
    closeOnEscape: function () {
        document.addEventListener('keydown',  (e) => {
            if (e.key === 'Escape' || e.code === 'Escape' || e.key === 'Esc') {
                if (typeof jsPanel.customCloseOnEscHandler === 'function') {
                    jsPanel.customCloseOnEscHandler (e);
                } else {
                    jsPanel.getPanels(function () {
                        return this.classList.contains('jsPanel');
                    }).some(function (item) {
                        if (item.options.closeOnEscape) {
                            item.close(null, true);
                            return true;
                        }

                        return false;
                    });
                }
            }
        }, false);
    }()
...
};
jsPanel.customCloseOnEscHandler  = () => {
// Custom implementation
}
```

**Describe alternatives you've considered**
1. An alternative would be to have a jsPanel.setCustomCloseOnEscHandler(handler) function that removes the original event listener and adds a new event listener with the event handler passed in the parameter.
2. Another option would be to make the event handler public so that we can remove the event listener and add our own.

**Context**
We would like to override the default implementation of the event handler because of our custom "Hide On Close" plugin that only hides the panels with display:none; and does not remove them from DOM. This causes the getPanels() in the event handler to also return hidden panels which may have higher z-index than the visible panels: jsPanel then tries to close the panel which is already closed (hidden).

I know this is a very specific issue we have, however, I can also imagine other scenarios in which it would be useful to have a custom implementation of the event handler.

Thank you for any assistance.

**Comment by Flyer53** [comment]
Hi there,

Wouldn't it be easier to allow to directly set **_option.closeOnEscape_** with your custom function? Like:
```javascript
jsPanel.create({
    closeOnEscape: (panel) => {
        // do whatever you want
        // a truthy return value would immediately stop processing other panels
        // a falsy return value would close the next panel in sequence
    }
});
```
This would be quite easy to implement without having to add/remove additional event handlers. And as far as I understand you it would do the job for you?

Regards,
Stefan

**Comment by Flyer53** [comment]
Ok, I implemented the feature **as described above** and you can [download v4.12.0-alpha](https://alpha.jspanel.de/downloads/jspanel-4.12.0-alpha.zip) to give it a try.

**Comment by akstek** [comment]
Thx Stefan, I am a co worker of Oilver and will test the provided Alpha and check back to you soon.

**Comment by akstek** [comment]
I have tested your provided solution and it works great for us.  Thank you for the fast response and implementation. Looking forward to the final v12 release.

**Comment by Flyer53** [comment]
Implemented with v4.12.0 release
